components/sst_importer/src/sst_writer.rs

```diff
@@ -380,4 +429,84 @@ mod tests {
         batch.set_ttl(10);
         assert!(w.write(batch).is_err());
     }
+
+    #[test]
+    fn test_raw_write_v1() {
+        let mut meta = SstMeta::default();
+        meta.set_uuid(Uuid::new_v4().as_bytes().to_vec());
+
+        let importer_dir = tempfile::tempdir().unwrap();
+        let cfg = Config::default();
+        let importer = SSTImporter::new(&cfg, &importer_dir, None, ApiVersion::V1).unwrap();
+        let db_path = importer_dir.path().join("db");
+        let db = new_test_engine(db_path.to_str().unwrap(), DATA_CFS);
+
+        let mut w = importer.new_raw_writer::<TestEngine>(&db, meta).unwrap();
+        let mut batch = RawWriteBatch::default();
+
+        let mut pair = Pair::default();
+        pair.set_key(b"k1".to_vec());
+        pair.set_value(b"v1".to_vec());
+        batch.set_pairs(vec![pair].into());
+        w.write(batch).unwrap();
+    }
+
+    #[test]
+    fn test_raw_write_invalid_key_mode() {
+        let mut meta = SstMeta::default();
+        meta.set_uuid(Uuid::new_v4().as_bytes().to_vec());
+
+        let importer_dir = tempfile::tempdir().unwrap();
+        let cfg = Config::default();
+        let importer = SSTImporter::new(&cfg, &importer_dir, None, ApiVersion::V2).unwrap();
+        let db_path = importer_dir.path().join("db");
+        let db = new_test_engine(db_path.to_str().unwrap(), DATA_CFS);
+
+        let mut w = importer.new_raw_writer::<TestEngine>(&db, meta).unwrap();
+        let mut batch = RawWriteBatch::default();
+        batch.set_ts(1);
+
+        // put an invalid key
+        let mut pair = Pair::default();
+        pair.set_key(b"k1".to_vec());
+        pair.set_value(b"short_value".to_vec());
+        let pairs = vec![pair];
+        batch.set_pairs(pairs.into());
+
+        assert!(w.write(batch).is_err());
+    }
+
+    #[test]
+    fn test_txn_write_v2() {
+        let mut meta = SstMeta::default();
```

Comment:
The following 7 lines are duplicated in almost all test cases. Suggest to move into a function.

Reply:
Done.